seagap/configs/main.py

```diff
@@ -0,0 +1,110 @@
+"""main.py
+
+The main configuration module containing base settings pydantic
+classes
+"""
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import yaml
+from pydantic import BaseSettings, Field
+from pydantic.fields import ModelField
+
+from .solver import Solver
+
+CONFIG_FILE = "config.yaml"
+
+
+def yaml_config_settings_source(settings: BaseSettings) -> Dict[str, Any]:
+    """
+    Read config settings form a local yaml file where the software runs
+
+    Parameters
+    ----------
+    settings : pydantic.BaseSettings
+        The base settings class
+
+    Returns
+    -------
+    dict
+        The configuration dictionary based on inputs from the yaml
+        file
+    """
+    encoding = settings.__config__.env_file_encoding
+    config_path = Path(CONFIG_FILE)
+    if config_path.exists():
+        # Only load config.yaml when it exists
+        return yaml.safe_load(config_path.read_text(encoding))
```

Comment:
How common is the 'no config file' use case? My only worry is that a misplaced or misnamed config may result in silently using default values. Would it be better to raise an error or at least log a warning?

Reply:
Gotcha, yeah, I think logging a warning that the config file was not found would be good.